cmd/agent/entrypoint.go

```diff
@@ -249,6 +257,68 @@ func (ep *Entrypoint) getReporterMetrics() map[string]interface{} {
 	return map[string]interface{}{"enabled-features": ep.cfg.EnabledFeatures}
 }
 
+func (ep *Entrypoint) supportHandler(rw http.ResponseWriter, r *http.Request) {
+	ep.mut.Lock()
+	cfg := ep.cfg
+	ep.mut.Unlock()
+
+	if cfg.DisableSupportBundle {
+		rw.WriteHeader(http.StatusNotFound)
+		_, _ = rw.Write([]byte("404 - support bundle generation is disabled"))
+		return
+	}
+	ep.sbmut.Lock()
+	defer ep.sbmut.Unlock()
+
+	duration := ep.srv.HTTPServer.WriteTimeout.Seconds()
+	if r.URL.Query().Has("duration") {
+		durationInt, err := strconv.Atoi(r.URL.Query().Get("duration"))
+		if err != nil {
+			http.Error(rw, fmt.Sprintf("duration value (in seconds) should be a positive integer: %s", err), http.StatusBadRequest)
+			return
+		}
+		if durationInt < 1 {
+			http.Error(rw, "duration value (in seconds) should be larger than 1", http.StatusBadRequest)
+			return
+		}
+		duration = float64(durationInt)
+		if duration > ep.srv.HTTPServer.WriteTimeout.Seconds() {
+			http.Error(rw, "duration value exceeds the server's write timeout", http.StatusBadRequest)
+			return
+		}
```

Comment:
pprof has an interesting way of doing this:

https://cs.opensource.google/go/go/+/refs/tags/go1.19.2:src/net/http/pprof/pprof.go;l=104-107

Maybe this is fine as-is, but getting the server from the context _might_ be more safe just in case `ep.srv.HTTPServer.WriteTimeout` gets mutated at read time.